docs/source/python/integration/python_r.rst

```diff
@@ -0,0 +1,312 @@
+    r_source("addthree.R")
+
+    # Get a reference to the addthree function
+    addthree = robjects.r["addthree"]
+
+    # Invoke the function
+    r = addthree(3)
+
+    # Access the returned value
+    value = r[0]
+    print(value)
+
+Running the ``addthree.py`` file will show how our Python code is able
+to access the ``R`` function and print the expected result:
+
+.. code-block:: bash
+
+    $ python addthree.py 
+    6.0
+
+If instead of passing around basic data types we want to pass around
+Arrow Arrays, we can do so relying on the ``rpy2-arrow`` module which
+implements rpy2 support for Arrow types.
+
+``rpy2`` can be installed through ``pip``:
+
+.. code-block:: bash
+
+    $ pip install rpy2-arrow
+
+``rpy2-arrow`` implements converters from pyarrow objects to r arrow objects,
+this is done without occurring into any data copy cost as it relies on the
+C Data interface.
+
+To pass to ``addthree`` a pyarrow array our ``addthree.py`` needs to be modified
+to enable ``rpy2-arrow`` converters and then pass the pyarrow array:
+
+.. code-block:: python
+
+    import rpy2.robjects as robjects
+    from rpy2_arrow.pyarrow_rarrow import (rarrow_to_py_array,
+                                           converter as arrowconverter)
+    from rpy2.robjects.conversion import localconverter
+
+    r_source = robjects.r["source"]
+    r_source("addthree.R")
+
+    addthree = robjects.r["addthree"]
+
+    import pyarrow
+
+    array = pyarrow.array((1, 2, 3))
+
+    # Enable rpy2-arrow converter so that R can receive the array.
+    with localconverter(arrowconverter):
+        r_result = addthree(array)
+
+    # The result of the R function will be an R Environment
+    # we can convert back the Environment to a pyarrow Array
+    # using the rarrow_to_py_array function
+    py_result = rarrow_to_py_array(r_result)
+    print("RESULT", type(py_result), py_result)
+
+Running the newly modified ``addthree.py`` should now properly execute
+the R function and print the resulting pyarrow Array:
+
+.. code-block:: bash
+
+    $ python addthree.py
+    RESULT <class 'pyarrow.lib.DoubleArray'> [
+      4,
+      5,
+      6
+    ]
+
+.. note::
+
+    Even though we sent an ``Int64Array`` to R, we end up with a
+    result as a ``DoubleArray``. That's due to the lack of native
+    support for 64 bits numbers in R and thus its use of doubles
+    to represent those.
+
+For additional information you can refer to
+`rpy2 Documentation <https://rpy2.github.io/doc/latest/html/index.html>`_
+and `rpy2-arrow Documentation <https://rpy2.github.io/rpy2-arrow/version/main/html/index.html>`_
+
+Invoking Python functions from R
+--------------------------------
+
+Exposing Python functions to R can be done through the ``reticulate``
+library. For example if we want to invoke :func:`pyarrow.compute.add` from
+R on an Array created in R we can do so importing ``pyarrow`` in R
+through ``reticulate``.
+
+A basic ``addthree.R`` script that invokes ``add`` to add ``3`` to
```

Comment:
Do we want to expand on this here or should we just point to the existing docs [on the R side](https://arrow.apache.org/docs/r/articles/python.html#using)?

cc @thisisnic  and @jonkeane

Reply:
I'm fine duplicating a bit in both places, but we should link as well

Reply:
Added link at the end of the paragraph near to the Reticulate one